File: web/src/entities/space/model/space.ts
export type SpaceEntityType = 'ooo' | 'ip' | 'fl' | 'smz';

export interface Space {
  id: number;
  title: string;
  link: string;
  logo?: string | null;
  description?: string | null;
  entity?: SpaceEntityType | null;
  director?: string | null;
  inn?: string | null;
  margin?: number | null;
  phone?: string | null;
  mail?: string | null;
  telegram?: string | null;
  country?: string | null;
  region?: string | null;
  city?: string | null;
  users?: number[];
  user?: number | null;
  created?: number;
  updated?: number;
}

export interface GetSpacesParams {
  id?: number | number[];
  link?: string;
  attached?: boolean;
  search?: string;
  limit?: number;
  offset?: number;
  attach?: boolean;
}

export interface SaveSpaceRequest {
  id?: number;
  link?: string;
  title?: string;
  logo?: string | null;
  description?: string | null;
  entity?: SpaceEntityType | null;
  director?: string | null;
  inn?: string | null;
  margin?: number | null;
  phone?: string | null;
  mail?: string | null;
  telegram?: string | null;
  country?: string | null;
  region?: string | null;
  city?: string | null;
  status?: number | null;
}

export interface SaveSpaceResponse {
  id: number;
  new: boolean;
  space: Space;
}

export interface GetSpacesResponse {
  spaces: Space[];
  count?: number | null;
}

File: api/app/models/space.py
from models import Base, Attribute


class Space(Base):
    _name = "spaces"
    _search_fields = {
        "title",
        "description",
        "mail",
        "telegram",
        "country",
        "region",
        "city",
    }

    title = Attribute(types=str)
    link = Attribute(types=str)
    logo = Attribute(types=str)
    description = Attribute(types=str, default="")
    entity = Attribute(types=str)
    director = Attribute(types=str)
    inn = Attribute(types=str)
    margin = Attribute(types=float, default=0.0)
    phone = Attribute(types=str)
    mail = Attribute(types=str)
    telegram = Attribute(types=str)
    country = Attribute(types=str)
    region = Attribute(types=str)
    city = Attribute(types=str)
    users = Attribute(types=list, default=list)
    user = Attribute(types=int)
    token = Attribute(types=str)
    status = Attribute(types=int, default=1)
